lohica-js-network/src/actions/searchActions.js

```diff
@@ -0,0 +1,39 @@
+import { searchRoute } from '../routes';
+export const SEARCH_RESULT = 'SEARCH_RESULT';
+export const SEARCH_RESULT_EMPTY = 'SEARCH_RESULT_EMPTY';
+
+export function setSearchResult(result) {
+    return {
+        type: SEARCH_RESULT,
+        payload: result
+    }
+}
+
+export function emptySearchResult() {
+    return {
+        type: SEARCH_RESULT_EMPTY
+    }
+}
+
+export function searchUsers (queueString, userId) {
+    const token = localStorage.getItem('tkn');
+    
+    return function(dispatch){
+        return fetch(`${searchRoute}?q=${queueString}`,{
```

Comment:
Create a common helper-class or method which would do XHR requests and use it everywhere you need

Reply:
did that for the whole project